rune/libenclave/internal/runtime/pal/pal_linux.go

```diff
@@ -118,14 +119,38 @@ func (pal *enclaveRuntimePal) Attest(isDCAP bool, isRA bool, spid string, subscr
 		return nil, nil
 	}
 
-	var targetInfo []byte
+	targetInfo := make([]byte, intelsgx.TargetinfoLength)
 	var err error
 
 	if isDCAP {
-		targetInfo, err = intelsgx.GetDCAPTargetInfo()
+		rf, err := os.Open("/opt/qe_targetinfo.bin")
```

Comment:
Just kindly FYI:
1) both epid and dcap can retrieve target info through protobuf, so it is required to implement dcap specific codes under the existing API GetQeTargetInfo().
2) I think GetSgxQeTargetInfo() itself can be smart enough to avoid using a flag to indicate which attestation type (epid or dcap) should be used. Of course explicitly specifying "isDCAP" is avoidable.
3) You can try to formally implement the codes in sgx-tools which calls GetQeTargetInfo() and send the formal PR with it :) This is preferred way than draining your energy on fighting against runelet and PAL :)

Reply:
4) It is pretty difficult to me to accept your implementation for PAL modification, but instead it is pretty easy to review the codes of sgx-tools and intelsgx package.

Reply:
> Just kindly FYI:
> 
> 1. Both epid and dcap can retrieve target info through protobuf, so it is required to implement dcap specific codes under the existing API GetQeTargetInfo().
> 2. I think GetSgxQeTargetInfo() itself can be smart enough to avoid using a flag to indicate which attestation type (epid or dcap) should be used. Of course explicitly specifying "isDCAP" is avoidable.
> 3. You can try to formally implement the codes in sgx-tools which calls GetQeTargetInfo() and send the formal PR with it. This is preferred way than draining your energy on fighting against runelet and PAL like current situation :)

Ok, use the work around method is help to me to achieve the get DCAP quote function. When successfully get dcap quote, I will revert this commit and push the format pr about sgx-tools.

Reply:
> 1. It is pretty difficult to me to accept your implementation for PAL modification in a short time, but instead it is pretty easy to review the codes of sgx-tools and intelsgx package. After all this PR is addressing sgx-tools not PAL.

This just helps me to get local report and achieve get quote function in host mode. When I achieve it, I will drop this pr  and reachieve rune attest in container mod.

Reply:
> Just kindly FYI:
> 
> 1. Both epid and dcap can retrieve target info through protobuf, so it is required to implement dcap specific codes under the existing API GetQeTargetInfo().
> 2. I think GetSgxQeTargetInfo() itself can be smart enough to avoid using a flag to indicate which attestation type (epid or dcap) should be used. Of course explicitly specifying "isDCAP" is avoidable.
> 3. You can try to formally implement the codes in sgx-tools which calls GetQeTargetInfo() and send the formal PR with it. This is preferred way than draining your energy on fighting against runelet and PAL like current situation :)

The /opt/target.bin is generated by sgx-tools qe-gen-target-info command in dcap mode. it is helpful to verify whether is correct or not. local report must be got inside enclave, use this work around to help me to ensure that skeleton get_local_report API is correct for dcap mode and can help me to verify sgx-tools gen-quote. For a new dcap developer, I think it is  helpful to develop and verify.

Reply:
Sounds like qe-gen-target-info command  is already completed right? I suggest you can submit it in the the formal way of development w/ CI/CD.

Reply:
> OK. Once qe-gen-target-info command with out-of-process mode is already completed, you can submit it in the the formal way of development w/ CI/CD.

Ok, work in progress.